File: java/v2.1/src/main/java/com/google/api/services/samples/dfareporting/reports/PatchReport.java
// Copyright 2014 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.services.samples.dfareporting.reports;

import com.google.api.services.dfareporting.Dfareporting;
import com.google.api.services.dfareporting.model.Report;
import com.google.api.services.samples.dfareporting.DfaReportingFactory;

/**
 * This example illustrates how to patch a standard report. Patching a report will modify only the
 * fields you specify.
 */
public class PatchReport {
  private static final String USER_PROFILE_ID = "INSERT_USER_PROFILE_ID_HERE";

  private static final String REPORT_ID = "INSERT_REPORT_ID_HERE";
  private static final String REPORT_NAME = "INSERT_REPORT_NAME_HERE";

  public static void runExample(Dfareporting reporting, long profileId, long reportId,
      String reportName) throws Exception {
    // Prepare the report object to patch.
    Report toPatch = new Report();
    // We only want to modify the report name.
    toPatch.setName(reportName);

    // Insert the report.
    Report report = reporting.reports().patch(profileId, reportId, toPatch).execute();

    System.out.printf("%s report with ID %d was successfully patched.%n", report.getType(),
        report.getId());
  }

  public static void main(String[] args) throws Exception {
    Dfareporting reporting = DfaReportingFactory.getInstance();

    long profileId = Long.parseLong(USER_PROFILE_ID);
    long reportId = Long.parseLong(REPORT_ID);

    runExample(reporting, profileId, reportId, REPORT_NAME);
  }
}

File: ruby/v2.6/reports/patch_report.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2016, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example illustrates how to patch a report

require_relative '../dfareporting_utils'

def patch_report(profile_id, report_id)
  # Authenticate and initialize API service.
  service = DfareportingUtils.get_service()

  # Create a report resource with the fields to patch.
  report = DfareportingUtils::API_NAMESPACE::Report.new({
    :criteria => DfareportingUtils::API_NAMESPACE::Report::Criteria.new({
      :date_range => DfareportingUtils::API_NAMESPACE::DateRange.new({
        :start_date => nil,
        :end_date => nil,
        :relative_date_range => 'YESTERDAY'
      })
    })
  })

  # Patch the report.
  result = service.patch_report(profile_id, report_id, report)

  # Display results.
  puts '%s report with ID %d and name "%s" has been patched.' %
      [result.type, result.id, result.name]
end

if __FILE__ == $0
  # Retrieve command line arguments.
  args = DfareportingUtils.get_arguments(ARGV, :profile_id, :report_id)

  patch_report(args[:profile_id], args[:report_id])
end
